app/apollo/resolvers/group.js

```diff
@@ -485,42 +554,52 @@ const groupResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, groupUuids, clusterId }, `${queryName} validating`);
-
-        await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context);
+        logger.info({req_id, user, org_id, clusterId, groupUuids}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'clusterId', clusterId );
         groupUuids.forEach( value => validateString( 'groupUuids', value ) );
 
-        const groups = await models.Group.find({ org_id, uuid: { $in: groupUuids } });
+        const groups = await getAllowedResources(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, null, groupUuids);
+        logger.info({req_id, user, org_id, clusterId, groupUuids}, `${queryName} retrieved allowed resources`);
+
         if (groups.length != groupUuids.length) {
           throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
         }
 
+        if (groups.length < 1) { throw new NotFoundError(context.req.t('None of the passed group uuids were found')); }
```

Comment:
This is a check that was not present in the original code, which I think would result in a change in behavior -- previously this `editClusterGroups` api could be used to remove a cluster from all groups by passing an empty-array `groupUuids` argument, but with this change it would instead throw a NotFoundError and fail.

Unless I'm missing something, please fix and also add a unit test for this scenario so that it can't accidentally regress in the future.

Reply:
Resolved